Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -0,0 +1,1230 @@
+                edgeCenters[10] = (boundsCorners[2] + boundsCorners[6]) * 0.5f;
+                edgeCenters[11] = (boundsCorners[3] + boundsCorners[7]) * 0.5f;
+            }
+        }
+        private Vector3 ClampScale(Vector3 scale, out bool clamped)
+        {
+            Vector3 finalScale = scale;
+            Vector3 maximumScale = initialScale * scaleMaximum;
+            clamped = false;
+
+            if (scale.x > maximumScale.x || scale.y > maximumScale.y || scale.z > maximumScale.z)
+            {
+                finalScale = maximumScale;
+                clamped = true;
+            }
+
+            Vector3 minimumScale = initialScale * scaleMinimum;
+
+            if (finalScale.x < minimumScale.x || finalScale.y < minimumScale.y || finalScale.z < minimumScale.z)
+            {
+                finalScale = minimumScale;
+                clamped = true;
+            }
+
+            return finalScale;
+        }
+        private Vector3 GetLinkDimensions()
+        {
+            float linkLengthAdjustor = wireframeShape == WireframeType.Cubic ? 2.0f : 1.0f - (6.0f * linkRadius);
+            return (currentBoundsSize * linkLengthAdjustor) + new Vector3(linkRadius, linkRadius, linkRadius);
+        }
+        private void ResetHandleVisibility()
+        {
+            bool isVisible;
+
+            //set balls visibility
+            if (balls != null)
+            {
+                isVisible = (wireframeOnly ? false : showRotateHandles);
+                for (int i = 0; i < ballRenderers.Count; ++i)
+                {
+                    ballRenderers[i].material = handleMaterial;
+                    ballRenderers[i].enabled = isVisible;
+                }
+            }
+
+            //set corner visibility
+            if (corners != null)
+            {
+                isVisible = (wireframeOnly ? false : showScaleHandles);
+                for (int i = 0; i < cornerRenderers.Count; ++i)
+                {
+                    cornerRenderers[i].material = handleMaterial;
+                    cornerRenderers[i].enabled = isVisible;
+                }
+            }
+
+            SetHiddenHandles();
+        }
+        private void ShowOneHandle(GameObject handle)
+        {
+            //turn off all balls
+            if (balls != null)
+            {
+                for (int i = 0; i < ballRenderers.Count; ++i)
+                {
+                    ballRenderers[i].enabled = false;
+                }
+            }
+
+            //turn off all corners
+            if (corners != null)
+            {
+                for (int i = 0; i < cornerRenderers.Count; ++i)
+                {
+                    cornerRenderers[i].enabled = false;
+                }
+            }
+
+            //turn on one handle
+            if (handle != null)
+            {
+                Renderer r = handle.GetComponent<Renderer>();
+                r.material = handleGrabbedMaterial;
+                r.enabled = true;
```

Comment:
Should we have also been keeping a list of handle renderers as well like we did above for corners and balls?

Reply:
maybe not- ill see if they need to be used and remove if not

Reply:
oops i misunderstood the original comment.

'handle' here isnt a type. if i got a reference to a ball or cornercube, it wouldnt be easily indexed into a list of renderers. it could be a cube primitive or a ball primiitive.